### PR TITLE
Optimize dependency declaration

### PR DESCRIPTION
**Pull request by darxriggs**
Extract declaring dependency versions required for upper bound management with the maven-enforcer-plugin to the `<dependencyManagement>` element. This just defines the versions but not the actual dependencies. Only direct dependencies should and are now declared in the `<dependencies>` element. This provides a better overview what is actually required. It also significantly reduces the size of the HPI file as less JAR files are packaged now.

### CODE REVIEW

**Comment by darxriggs** [comment]
That's pretty strange that locally all tests pass and on GitHub the enforcer fails. I'll have to look into it.

**Comment by jeffpearce** [comment]
I can repro running `mvn clean install`

File: pom.xml [hunk 1]
```diff
@@ -50,158 +50,82 @@
...
+                <scope>test</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
     <dependencies>
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
             <version>2.8.5</version>
         </dependency>
```
Comment (jeffpearce) [inline pom.xml]:
```suggestion
        </dependency>
        <dependency>
            <groupId>org.apache.maven.doxia</groupId>
            <artifactId>doxia-sink-api</artifactId>
            <version>1.1.2</version>
        </dependency>
        <dependency>
            <groupId>org.codehaus.plexus</groupId>
            <artifactId>plexus-container-default</artifactId>
            <version>1.0-alpha-30</version>
        </dependency>
        <dependency>
            <groupId>org.codehaus.plexus</groupId>
            <artifactId>plexus-utils</artifactId>
            <version>3.0.24</version>
        </dependency>
```

Comment (jeffpearce) [reply pom.xml]: This fixes it locally for me

**Comment by darxriggs** [comment]
For reference: The failing tests are due to different Maven versions used - locally: 3.6.2, ci.jenkins.io: 3.5.4.

It seems that newer Maven versions have an improved dependency resolution. I used Docker locally to verify this.

```$ docker run -it --rm -v "$HOME/.m2":/root/.m2 -v "$(pwd)":/usr/src/work -w /usr/src/work maven:3.5.4-jdk-8 mvn clean verify```

```$ docker run -it --rm -v "$HOME/.m2":/root/.m2 -v "$(pwd)":/usr/src/work -w /usr/src/work maven:3.6.2-jdk-8 mvn clean verify```

Therefore I'll re-add some dependency versions to make the plugin at least build with Maven 3.5.4.

**Comment by darxriggs** [comment]
@jeffpearce now it's ready to merge.

**Pull request merged**